### PR TITLE
Convert more uses of `String` to `Text`.

### PR DESCRIPTION
**Pull request by brianhuffman**
This helps with #1269, removing several occurrences of `Text.unpack`.

### CODE REVIEW

File: src/SAWScript/Crucible/LLVM/Builtins.hs [hunk 1]
```diff
@@ -1439,7 +1440,7 @@ setupArg sc sym ecRef tp = do
     freshGlobal cty sc_tp =
       do ecs <- readIORef ecRef
          let len = Seq.length ecs
-         ec <- scFreshEC sc ("arg_"++show len) sc_tp
+         ec <- scFreshEC sc ("arg_" <> Text.pack (show len)) sc_tp
```
Comment (RyanGlScott) [inline src/SAWScript/Crucible/LLVM/Builtins.hs:1443]: There are two calls to `Text.pack` here, one of which is implicit in the use of `OverloadedStrings`. This can be shaved down to just one:

```suggestion
         ec <- scFreshEC sc (Text.pack ("arg_" ++ show len)) sc_tp
```

File: src/SAWScript/Crucible/LLVM/Skeleton/Builtins.hs [hunk 2]
```diff
@@ -202,7 +202,7 @@ buildArg arg idx
         $ arg ^. argSkelType . typeSkelLLVMType
       pure (Just val, Nothing, arg ^. argSkelName)
   where
-    ident = maybe ("arg" <> show idx) Text.unpack $ arg ^. argSkelName
+    ident = maybe ("arg" <> Text.pack (show idx)) id $ arg ^. argSkelName
```
Comment (RyanGlScott) [inline src/SAWScript/Crucible/LLVM/Skeleton/Builtins.hs:205]:
```suggestion
    ident = maybe (Text.pack ("arg" ++ show idx)) id $ arg ^. argSkelName
```

File: src/SAWScript/Crucible/LLVM/Skeleton/Builtins.hs [hunk 3]
```diff
@@ -237,7 +237,7 @@ rebuildArg (arg, prearg) idx
             -> LLVM.Array (fromIntegral $ s ^. sizeGuessElems) pt
           | otherwise -> pt
         _ -> pt
-      ident = maybe ("arg" <> show idx) Text.unpack nm
+      ident = maybe ("arg" <> Text.pack (show idx)) id nm
```
Comment (RyanGlScott) [inline src/SAWScript/Crucible/LLVM/Skeleton/Builtins.hs:240]:
```suggestion
      ident = maybe (Text.pack ("arg" ++ show idx)) id nm
```

File: src/SAWScript/Crucible/LLVM/Builtins.hs (hunk 1 above)
Comment (brianhuffman) [reply src/SAWScript/Crucible/LLVM/Builtins.hs:1443]: I had assumed that any calls to `Text.pack` arising from `OverloadedStrings` would be constant-folded away by the compiler; is this not the case? I was actually thinking that the obvious next improvement for this line would be to find a replacement for `pack . show`.

Comment (RyanGlScott) [reply src/SAWScript/Crucible/LLVM/Builtins.hs:1443]: I thought I recalled `pack` being a performance bottleneck, but now that I look closer, that's for [`ByteString`'s `pack`](https://hackage.haskell.org/package/bytestring-0.11.1.0/docs/Data-ByteString-Char8.html#v:pack), not [`Text`'s `pack`](https://hackage.haskell.org/package/text-1.2.4.1/docs/Data-Text.html#v:pack).

Both versions will constant-fold the `"arg_"` part, so I suppose it's just a question of whether it's faster to append two strings and `pack` them at the end, or to `pack` two strings and them append them in the end. But for strings this small, it seems very unlikely to matter.

**tl;dr** Go with whichever style looks better to you.

**Review by RyanGlScott — APPROVED** [review]

File: src/SAWScript/Crucible/LLVM/Builtins.hs (hunk 1 above)
Comment (brianhuffman) [reply src/SAWScript/Crucible/LLVM/Builtins.hs:1443]: I like the idea of minimizing the number of subexpressions of type `String`, so if the performance is about the same, I'll just keep it as is.
